assets/RNAseq_report.Rmd

```diff
@@ -364,26 +897,221 @@ This visualizes how similar the samples are to each other according to gene expr
 Two samples have more similar gene expression if their distance is closer to 0 (darker blue).
 The original plot can be downloaded [here](./differential_gene_expression/plots/Heatmaps_of_distances.pdf).
 
-```{r, echo=F, out.width="80%", dpi=1200, fig.align='center'}
-knitr::include_graphics(paste0(wd,"/differential_gene_expression/plots/Heatmaps_of_distances.svg"))
+```{r heatmap, echo=FALSE, warning=FALSE, results='hide'}
+##################  SAMPLE DISTANCES HEATMAP ##################
+# Sample distances
+sampleDists <- dist(t(assay(if (params$rlog) rld else vsd)))
+sampleDistMatrix <- as.matrix(sampleDists)
+colours = colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
+pdf("differential_gene_expression/plots/Heatmaps_of_distances.pdf")
+par(oma=c(3,3,3,3))
+dist_heatmap <- pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, col=colours,fontsize=10)
+dev.off()
+svg("differential_gene_expression/plots/Heatmaps_of_distances.svg")
+pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, col=colours,fontsize=10)
+dev.off()
+png("differential_gene_expression/plots/Heatmaps_of_distances.png")
+pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, col=colours,fontsize=10)
+dev.off()
+#knitr::include_graphics(paste0(wd,"/differential_gene_expression/plots/Heatmaps_of_distances.svg"))
+```
+
+```{r heatmap_plot, echo=F, out.width="80%", dpi=1200, fig.align='center'}
+dist_heatmap
 ```
 
 # Differential expression analysis
 
 To perform differential gene expression analysis, the gene expression data between two sample groups is compared.
 These comparisons are called _contrasts_. For this analysis, the following contrasts or comparisons were made:
 
-```{r contrasts, echo=FALSE, message=FALSE, bootstrap.show.message=FALSE, include=FALSE}
-df_contr <- read.table(params$path_contrasts, header=FALSE, sep="\t")
-contrnames = df_contr$V1
+<!-- contrast calculation start -->
```

Comment:
The idea of the Rmarkdown report, is that each section contains the code that is needed to reproduce the plots. Could you please move the code that calculates the DESEq2 objects here? That would be the section that it belongs to

Reply:
I cannot move the block unless I restructure the actual report. In the block above the code you marked, the heatmaps are put into the report, and above, the PCA is embedded, and both of these blocks require the deseq results